Update roku.markdown for Roku async update

dozminic opened the pull request:
## Proposed change
Starting with PR #35104 for the Roku component, the code now supports using the Roku channel name in addition to the appID. Additionally, the link to the Roku dev documentation has changed. This updates the documentation to address those items.

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [X] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: #35104
- Link to parent pull request in the Brands repository: N/A
- This PR fixes or closes issue: N/A

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

dozminic committed:
Update roku.markdown for Roku async update

Starting with PR #35104 for the Roku component, the code now supports using the Roku channel name in addition to the appID. Additionally, the link to the Roku dev documentation has changed.

dozminic committed:
Fixed example of Roku select source by name

"Amazon Prime" is not a valid Roku channel. Corrected to "Prime Video"

frenck commented on source/_integrations/roku.markdown:
````diff
@@ -93,7 +93,16 @@ data:
 ## Media Player
 
-When the Home Assistant Roku integration is enabled and a Roku device has been configured, in the Home Assistant UI the Roku media player will show a listing of the installed channels, or apps, under “source”. Select one and it will attempt to launch the channel on your Roku device. This action can also be automated, but it requires you to acquire an extra piece of information; the ```appID``` fo ...
+When the Home Assistant Roku integration is enabled and a Roku device has been configured, in the Home Assistant UI the Roku media player will show a listing of the installed channels, or apps, under “source”. Select one and it will attempt to launch the channel on your Roku device. This action can also be automated. Beginning with Home Assistant Core 0.110, channels can be launched by ```name``` ...
````
Please remove the version number reference. Our documentation should match the latest release.
Adding in specific version exceptions tend to stick around needed long and make the documentation look outdated/unmaintained.

frenck commented on source/_integrations/roku.markdown:
````diff
@@ -93,7 +93,16 @@ data:
+  service: media_player.select_source
+```
+
+Prior to Home Assistant Core 0.110, only the ```appID``` for the channel specific to your Roku can be used for `source:` Although this information is gathered by the Roku integration, at the moment it is not exposed to the end user. This item might be added in a future release. For now though, you can easily get the information yourself. All you need to do is a simple GET API call on the same net ...
````
Same as above, remove the version reference and make it reflect the current state.
Some additional typo's:

```suggestion
Prior to Home Assistant Core 0.110, only the ```appID``` for the channel-specific to your Roku can be used for `source:` Although this information is gathered by the Roku integration, at the moment it is not exposed to the end-user. This item might be added in a future release. For now, you can easily get the information yourself. All you need to do is a simple GET API call on the same network as your device.
```

frenck reviewed (changes requested):
Thanks for the PR @dozminic!

Some small comments, could you take a look? Thanks! 👍

dozminic committed:
Removed HA version references

Removed references to HA 0.110, per frenck's notes on PR #13629. Fixed minor typos.

dozminic committed:
Fixed appID info and clarified GET request process

Removed statement that appID is specific to an individuals Roku. From what I've been able to find out, the appID remains constant. Technically, you could find the appID by going to the Roku Channel Store and looking at the channel's URL (the appID is contained in it at this time), but sometimes the search won't return results for a channel that exists. Fetching via GET request (i.e. loading the page in your browser) is easier and more reliable a method.

dozminic commented:
@frenck I have made several updates which I believe satisfy your comments. If you have more, let me know! Turns out, the appID isn't specific to a user's Roku from what I can tell. That information has been removed. I also added information on how to easily make a GET request to the Roku API endpoint (i.e. just use your browser).

dozminic committed:
Typo fix

dozminic committed:
Additional typo/grammar fix

frenck commented on source/_integrations/roku.markdown:
````diff
@@ -93,21 +93,32 @@ data:
+  service: media_player.select_source
+```
+
+Alternatively, the ```appID``` for the channel can be used for `source:` Although this information is gathered by the Roku integration, at the moment it is not exposed to the end-user. This item might be added in a future release. For now, you can easily get the information yourself. All you need to do is a simple GET API call on the same network as your device.
````
Small change to use single backticks for inline code fencing.

Removed "simple", that term should be avoided in general, as it might not be a simple subject for everybody and thus condescending.

```suggestion
Alternatively, the `appID` for the channel can be used for `source:` Although this information is gathered by the Roku integration, at the moment it is not exposed to the end-user. This item might be added in a future release. For now, you can easily get the information yourself. All you need to do is a GET API call on the same network as your device.
```

frenck commented on source/_integrations/roku.markdown:
````diff
@@ -93,21 +93,32 @@ data:
 ## Media Player
 
-When the Home Assistant Roku integration is enabled and a Roku device has been configured, in the Home Assistant UI the Roku media player will show a listing of the installed channels, or apps, under “source”. Select one and it will attempt to launch the channel on your Roku device. This action can also be automated, but it requires you to acquire an extra piece of information; the ```appID``` fo ...
+When the Home Assistant Roku integration is enabled and a Roku device has been configured, in the Home Assistant UI the Roku media player will show a listing of the installed channels, or apps, under “source”. Select one and it will attempt to launch the channel on your Roku device. This action can also be automated. Channels can be launched by ```name``` using a configuration similar to the one  ...
````
```suggestion
When the Home Assistant Roku integration is enabled and a Roku device has been configured, in the Home Assistant UI the Roku media player will show a listing of the installed channels, or apps, under “source”. Select one and it will attempt to launch the channel on your Roku device. This action can also be automated. Channels can be launched by `name` using a configuration similar to the one below:

```

frenck commented on source/_integrations/roku.markdown:
````diff
@@ -93,21 +93,32 @@ data:
 ```
 
-More details can be found on the [Roku dev pages](https://developer.roku.com/docs/developer-program/discovery/external-control-api.md)
+The simplest method of performing the GET request is to open `http://ROKU_IP:8060/query/apps` in your web browser of choice. The Roku will return an XML-formatted list of available channels, including their full name and appID. 
+
+More details can be found on the [Roku dev pages](https://developer.roku.com/docs/developer-program/debugging/external-control-api.md)
````
```suggestion
The method of performing the GET request is to open `http://ROKU_IP:8060/query/apps` in your web browser of choice. The Roku will return an XML-formatted list of available channels, including their full name and appID. 

More details can be found on the [Roku dev pages](https://developer.roku.com/docs/developer-program/debugging/external-control-api.md)
```

dozminic committed:
Additional phrasing and formatting corrections

frenck reviewed (approved):
Thanks, @dozminic 👍
